### DISCUSSION TITLE
2 Point correlation function under the hood

### DISCUSSION THREAD

**Discussion by ylracnna** [PMEAL/porespy#602] [Q&A]
I am interested in using the 2 point correlation function in the metrics submodule. However, I'm assuming that the function has to do some rudimentary segmentation to turn a greyscale image into black and white to perform analysis. Does anyone know what segmentation it is performing/what methods it's using?

**Comment by jgostick** [comment]
Yes, it must be segmented into a binary image with ``True`` for the phase of interest.  However, we do not do this for you.  It's a very tricky business and varies on a case by case basis, so we leave that in the hands of the user.

**Reply by ylracnna** [reply]
Understood. However, currently, if I feed a grayscale image through the function, I get a response. What is happening when I do that?

**Reply by jgostick** [reply]
The algorithm uses a fourier transform to compute the correlation, so I suppose there is a response when a grey scale images is cross-correlated with itself, but I'm not sure what this means.  We used to have a 'brute force' method which literally counted the number of 'hits' within a template centered on each pixel/voxel.  That method would also have given a result, but all ones, since any pixel with a value greater than 0 would have been considered in the void phase, and this would have basically found that ALL pixels are in the void phase.  In any event, here is a tutorial that I like which emphasizes the care that needs to be taken when doing segmentation: https://emmanuelle.github.io/a-tutorial-on-segmentation.html
